### ISSUE TITLE
Strict setting does not handle ternary operator correctly.

### ISSUE THREAD

**Issue by DeborahK** [Microsoft/TypeScript#19880]
**TypeScript Version:**  2.7.0-dev.201xxxxx

                (params.get('title') ?
                    movie.title.toLocaleLowerCase().indexOf(params.get('title').toLocaleLowerCase()) !== -1 : true)

```ts
// A *self-contained* demonstration of the problem follows...
// Test this by running `tsc` on the command-line, rather than through another build tool such as Gulp, Webpack, etc.
```

**Expected behavior:**
The above line of code should be fine ... not flagged.

**Actual behavior:**
The above line of code generates an "Object is possibly 'null'." warning on "params.get('title')" in the first expression (2nd argument) of the ternary operator. Since the condition uses the same expression, the expression cannot be null in this case. If it were, the third expression would be executed.

**Comment by weswigham** [comment]
Nothing precludes `params.get('title')` from returning a different result each call, so we can't make the assumption that it's still not null. If you're sure that the implementation of the `get` function is consistent, then you should use the non-null assertion operator, postfix-`!`, on the second usage. Alternatively, extract `params.get('title')` to a local and reuse the local in both places.

**Comment by DeborahK** [comment]
Thank you for your quick response. Good point. Though highly unlikely that the parameters would change through this bit of code, it cannot be said with certainty that it won't.
This is part of a much larger piece of code that is checking several parameters ... so extracting to local variables would make it longer/more difficult to reason about.
I'll use the non-null assertion operator. 
Thanks again!
